packages/files-ui/src/Components/Modules/FileBrowsers/views/FileSystemItem/SharedFolderRow.tsx

```diff
@@ -110,19 +113,99 @@ const useStyles = makeStyles(({ breakpoints, constants, palette }: CSFTheme) =>
 
 interface Props {
   bucket: BucketKeyPermission
-  onFolderClick: (e?: React.MouseEvent) => void
-  menuItems: IMenuItem[]
-  isEditing: boolean
-  setIsEditing: (isEditing: boolean) => void
   handleRename: (bucket: BucketKeyPermission, newName: string) => void
+  openSharedFolder: (bucketId: string) => void
+  onEditSharedFolder: () => void
+  handleDeleteSharedFolder: () => void
 }
 
-const SharedFolderRow = ({ bucket, onFolderClick, menuItems, isEditing, setIsEditing, handleRename }: Props) => {
+const SharedFolderRow = ({ bucket, handleRename, openSharedFolder, handleDeleteSharedFolder, onEditSharedFolder }: Props) => {
   const classes = useStyles()
   const { name, size } = bucket
+
   const { desktop } = useThemeSwitcher()
+  const [isRenaming, setIsRenaming] = useState(false)
+
   const isOwner = useMemo(() => bucket.permission === "owner", [bucket.permission])
 
+  const menuItems: IMenuItem[] = isOwner
+    ? [{
+      contents: (
+        <>
+          <EditSvg className={classes.menuIcon} />
+          <span data-cy="menu-rename">
+            <Trans>Rename</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => setIsRenaming(true)
+    },
+    {
+      contents: (
+        <>
+          <UpdateSvg className={classes.menuIcon} />
+          <span data-cy="menu-edit">
+            <Trans>Edit</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => onEditSharedFolder()
+    }, {
+      contents: (
+        <>
+          <DeleteSvg className={classes.menuIcon} />
+          <span data-cy="menu-delete">
+            <Trans>Delete</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => handleDeleteSharedFolder()
+    }]
+    : [{
+      contents: (
+        <>
+          <DeleteSvg className={classes.menuIcon} />
+          <span data-cy="menu-delete">
+            <Trans>Leave</Trans>
+          </span>
+        </>
+      ),
+      onClick: () => handleDeleteSharedFolder()
+    }]
+
+
+  const onSingleClick = useCallback(
+    () => {
+      if (desktop) {
+        // on desktop 
+      } else {
+        // on mobile
+        openSharedFolder(bucket.id)
+      }
+    },
```

Comment:
I know it's a copy/paste from me, but I feel we can get rid of this empty if.
```suggestion
      if (!desktop) {
        // on mobile
        openSharedFolder(bucket.id)
      }
    },
```